shapely/strtree.py

```diff
@@ -34,317 +34,185 @@ class STRtree:
 
-    An index is initialized from a sequence of geometry objects and
-    optionally a sequence of items. If items are not provided, the
-    indices of the geometry sequence will be used instead.
+    All operations return indices of the input geometries.  These indices
+    can be used to index into anything associated with the input geometries,
+    including the input geometries themselves, or custom items stored in
+    other object of the same length as the geometries.
 
-    Stored items and corresponding geometry objects can be spatially
-    queried using another geometric object.
+    Any mixture of geometry types may be stored in the tree.
 
     Parameters
     ----------
     geoms : sequence
         A sequence of geometry objects.
-    items : sequence, optional
-        A sequence of objects which typically serve as identifiers in an
-        application. This sequence must have the same length as geoms.
     node_capacity : int, default 10
         The maximum number of child nodes per parent node in the tree.
 
-    Examples
-    --------
-    Creating an index of polygons:
-
-    >>> from shapely.strtree import STRtree
-    >>> from shapely.geometry import Polygon
-    >>>
-    >>> polys = [Polygon(((0, 0), (1, 0), (1, 1))),
-    ...          Polygon(((0, 1), (0, 0), (1, 0))),
-    ...          Polygon(((100, 100), (101, 100), (101, 101)))]
-    >>> tree = STRtree(polys)
-    >>> query_geom = Polygon(((-1, -1), (2, 0), (2, 2), (-1, 2)))
-    >>> result = tree.query(query_geom)
-    >>> polys[0] in result
-    True
-    >>> polys[1] in result
-    True
-    >>> polys[2] in result
-    False
-
-    >>> import shapely
-    >>> tree = shapely.STRtree(shapely.points(np.arange(10), np.arange(10)))
-    >>> # Query geometries that overlap envelope of input geometries:
-    >>> tree.query(shapely.box(2, 2, 4, 4)).tolist()
-    [2, 3, 4]
-    >>> # Query geometries that are contained by input geometry:
-    >>> tree.query(shapely.box(2, 2, 4, 4), predicate='contains').tolist()
-    [3]
-    >>> # Query geometries that overlap envelopes of ``geoms``
-    >>> tree.query_bulk([shapely.box(2, 2, 4, 4), shapely.box(5, 5, 6, 6)]).tolist()
-    [[0, 0, 0, 1, 1], [2, 3, 4, 5, 6]]
-    >>> tree.nearest([shapely.points(1,1), shapely.points(3,5)]).tolist()  # doctest: +SKIP
-    [[0, 1], [1, 4]]
-
     References
     ----------
     .. [1] Leutenegger, Scott T.; Edgington, Jeffrey M.; Lopez, Mario A.
        (February 1997). "STR: A Simple and Efficient Algorithm for
        R-Tree Packing".
        https://ia600900.us.archive.org/27/items/nasa_techdoc_19970016975/19970016975.pdf
-
     """
 
     def __init__(
         self,
         geoms: Iterable[BaseGeometry],
-        items: Iterable[Any] = None,
         node_capacity: int = 10,
     ):
         # Keep references to geoms
         self.geometries = np.asarray(geoms, dtype=np.object_)
+        # prevent modification
+        self.geometries.flags.writeable = False
```

Comment:
This is something to check against downstream implementations.  In theory, nobody _should_ be modifying these and expecting things to not break (we guard against segfaults related to doing so), but this sets a flag to help prevent them from attempting to modify the geometries.